tests/python/gaia-ui-tests/gaiatest/gaia_test.py

```diff
@@ -993,8 +996,10 @@ def cleanup_gaia(self, full_reset=True):
             # reset to home screen
             self.device.touch_home_button()
 
-        # restore prefs from testvars
-        for name, value in self.testvars.get('prefs', {}).items():
+        defaults = DEFAULT_PREFS.copy()
+        defaults.update(self.testvars.get('prefs', {}))
+        defaults = self.modify_settings(defaults)
```

Comment:
I also wonder if we should be setting these preferences earlier in the `setUp` (for example, before we enable WiFi) to avoid any potential conflicts.